diff --git a/vendor/quiver/src/index.html b/vendor/quiver/src/index.html
index df93221..1074e00 100644
--- a/vendor/quiver/src/index.html
+++ b/vendor/quiver/src/index.html
@@ -108,7 +108,11 @@
         // corresponding URL. An array containing parameters to `ignore` is used to delete unwanted
         // query parameters.
         function reconstruct_URL(query_data, ignore) {
-            const URL_prefix = new URL(window.location.href);
+            const proto = window.location.protocol;
+            const URL_prefix = new URL(
+                (proto === "vscode-webview:" || proto === "vscode-webview-resource:")
+                    ? "https://q.uiver.app/" : window.location.href
+            );
             URL_prefix.search = "";
             URL_prefix.hash = "";
             return `${URL_prefix}#${
diff --git a/vendor/quiver/src/quiver.mjs b/vendor/quiver/src/quiver.mjs
index 63fdb4d..462f00d 100644
--- a/vendor/quiver/src/quiver.mjs
+++ b/vendor/quiver/src/quiver.mjs
@@ -1269,7 +1269,13 @@ QuiverImportExport.base64 = new class extends QuiverImportExport {
     export(quiver, settings, options) {
         // Remove the query string and fragment identifier from the current URL and use that as a
         // base.
-        const URL_prefix = window.location.href.replace(/\?.*$/, "").replace(/#.*$/, "");
+        const URL_prefix = (() => {
+            const proto = window.location.protocol;
+            if (proto === "vscode-webview:" || proto === "vscode-webview-resource:") {
+                return "https://q.uiver.app/";
+            }
+            return window.location.href.replace(/\?.*$/, "").replace(/#.*$/, "");
+        })();
 
         if (quiver.is_empty()) {
             // No need to have an encoding of an empty quiver; we'll just use the URL directly.
